### ISSUE TITLE
str.format no longer accepts unpacked defaultdict for default values

### ISSUE THREAD

**Issue by ff164935-b039-4486-b258-64b1d98020a6** [python/cpython#74775]
BPO | [30590](https://bugs.python.org/issue30590)
--- | :---
Nosy | @ericvsmith, @serhiy-storchaka, @carmenbianca

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-06-07.17:49:40.750>
created_at = <Date 2017-06-07.16:39:08.413>
labels = ['invalid', 'library']
title = 'str.format no longer accepts unpacked defaultdict for default values'
updated_at = <Date 2017-06-07.17:49:40.749>
user = 'https://github.com/carmenbianca'
```

bugs.python.org fields:
```python
activity = <Date 2017-06-07.17:49:40.749>
actor = 'serhiy.storchaka'
assignee = 'none'
closed = True
closed_date = <Date 2017-06-07.17:49:40.750>
closer = 'serhiy.storchaka'
components = ['Library (Lib)']
creation = <Date 2017-06-07.16:39:08.413>
creator = 'carmenbianca'
dependencies = []
files = []
hgrepos = []
issue_num = 30590
keywords = []
message_count = 2.0
messages = ['295353', '295364']
nosy_count = 3.0
nosy_names = ['eric.smith', 'serhiy.storchaka', 'carmenbianca']
pr_nums = []
priority = 'normal'
resolution = 'not a bug'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue30590'
versions = ['Python 3.6']
```

</p></details>

**Comment by ff164935-b039-4486-b258-64b1d98020a6** [comment]
As brief as can be, the following script works in Python 3.4 and Python 3.5:

```py
    from collections import defaultdict

    mydict = defaultdict(lambda: 'default')

    print('{foo}'.format(**mydict))
```


And prints "default".

In Python 3.6, you get a KeyError for foo.

**Comment by serhiy-storchaka** [comment]
There was a bug (see bpo-18531). It was fixed in 3.6.

Use str.format_map() for formatting with arbitrary mapping.
